SliceTracker/SliceTracker.py

```diff
@@ -1504,6 +1524,51 @@ def getAndSelectTargetFromTable(self):
     if modelIndex:
       self.targetTable.clicked(modelIndex)
 
+  def updateDisplacementChartTargetSelectorTable(self):
+    self.targetDisplacementChart.DisplacementChartTargetSelector.blockSignals(True)
+    self.targetDisplacementChart.DisplacementChartTargetSelector.clear()
+    if self.targetTableModel.rowCount():
+      rows = self.targetTableModel.rowCount()
+      for i in range(rows):
+        self.targetDisplacementChart.DisplacementChartTargetSelector.addItem(self.targetTableModel.targetList.GetNthFiducialLabel(i))
+    self.targetDisplacementChart.DisplacementChartTargetSelector.setCurrentIndex(-1)
+    self.targetDisplacementChart.DisplacementChartTargetSelector.blockSignals(False)
+
+  def calculateTargetDisplacement(self, prevTargets, currTargets):
+    prevPos = [0.0, 0.0, 0.0]
+    currPos = [0.0, 0.0, 0.0]
+    for i in range(0, currTargets.GetNumberOfFiducials()):
+      if currTargets.GetNthFiducialLabel(i) == self.targetDisplacementChart.DisplacementChartTargetSelector.currentText:
+        currTargets.GetNthFiducialPosition(i, currPos)
+        prevTargets.GetNthFiducialPosition(i, prevPos)
+        displacement = [currPos[0] - prevPos[0], currPos[1] - prevPos[1], currPos[2] - prevPos[2]]
+        return displacement
+
+  def onDisplacementChartTargetSelectionChanged(self):
+    if self.targetDisplacementChart.DisplacementChartTargetSelector.currentText != self.targetDisplacementChart.lastSelectedTarget:
+      self.targetDisplacementChart.resetAndInitializeChart()
+      if self.targetDisplacementChart.DisplacementChartTargetSelector.currentText:
+        self.targetDisplacementChart.lastSelectedTarget = self.targetDisplacementChart.DisplacementChartTargetSelector.currentText
+      results = sorted(self.registrationResults.getResultsAsList(), key=lambda s: s.seriesNumber)
+      iter1 = 0
+      iter2 = 1
+      while iter2 < len(results):
+        if results[iter1].approved and results[iter2].approved:
+          prevTargets = results[iter1].approvedTargets
+          currTargets = results[iter2].approvedTargets
+          displacement = self.calculateTargetDisplacement(prevTargets, currTargets)
+          self.targetDisplacementChart.addPlotPoints([displacement])
+          iter1 = iter2
+          iter2 += 1
+        elif results[iter1].approved:
+          iter2 += 1
+        elif results[iter2].approved:
+          iter1 = iter2
+          iter2 += 1
+        else:
+          iter2 += 1
+          iter1 += 1
+
```

Comment:
Instead of writing:
```
iter1 = 0
      iter2 = 1
      while iter2 < len(results):
        if results[iter1].approved and results[iter2].approved:
          prevTargets = results[iter1].approvedTargets
          currTargets = results[iter2].approvedTargets
          displacement = self.calculateTargetDisplacement(prevTargets, currTargets)
          self.targetDisplacementChart.addPlotPoints([displacement])
          iter1 = iter2
          iter2 += 1
        elif results[iter1].approved:
          iter2 += 1
        elif results[iter2].approved:
          iter1 = iter2
          iter2 += 1
        else:
          iter2 += 1
          iter1 += 1
```

You could write:
```
prevIndex = 0
for currentIndex, currResult in enumerate(results, 1):
  if results[prevIndex].approved and currResult.approved:
    prevTargets = results[prevIndex].approvedTargets
    currTargets = currResult.approvedTargets
    displacement = self.calculateTargetDisplacement(prevTargets, currTargets)
    self.targetDisplacementChart.addPlotPoints([displacement])
    prevIndex = currentIndex
  elif currResult.approved:
    prevIndex = currentIndex
  else:
    prevIndex += 1
```